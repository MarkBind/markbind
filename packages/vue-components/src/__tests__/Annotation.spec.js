import { mount } from '@vue/test-utils';
import { Dropdown } from 'floating-vue';
import Annotation from '../annotations/Annotate.vue';
import AnnotationPoint from '../annotations/AnnotatePoint.vue';

const DEFAULT_STUBS = {
  'a-point': AnnotationPoint,
  'v-popover': Dropdown,
};

const ANNOTATION_POINTS = `
  <a-point x="25%" y="25%" content="25% from the left and 25% from the top" />
  <a-point x="50%" y="25%" content="50% from the left and 25% from the top"  size="60"/>
  <a-point x="75%" y="25%" content="75% from the left and 25% from the top"  header="This has a header"/>
  <a-point x="33%" y="50%" content="33% from the left and 50% from the top"  color="red"/>
  <a-point x="66%" y="50%" content="66% from the left and 50% from the top"  opacity="0.7"/>
  <a-point x="25%" y="75%" content="25% from the left and 75% from the top" label="1"/>
  <a-point x="50%" y="75%" content="50% from the left and 75% from the top" 
    textColor="white" color="black" label="2" opacity="1"/>
  <a-point x="75%" y="75%" content="75% from the left and 75% from the top"  fontSize="30" label="3"/>
`;

const CUSTOMISED_ANNOTATION_POINTS = `
  <a-point x="25%" y="25%" content="25% from the left and 25% from the top">
    <div style="color:red">Hi</div>
  </a-point>
  <a-point x="75%" y="75%" content="75% from the left and 75% from the top">
  <span class="badge bg-primary">Primary</span>
  </a-point>
`;

const MARKDOWN_ANNOTATION_POINTS = `
  <a-point x="25%" y="25%" content="[link](https://markbind.org/userGuide/gettingStarted.html)" />
  <a-point x="50%" y="50%" header="### Header" />
  <a-point x="75%" y="75%" label=":rocket:" />
`;

describe('Annotation', () => {
  test('with different visual annotation points', async () => {
    const wrapper = mount(Annotation, {
      propsData: {
        src: './annotateSampleImage.png',
      },
      slots: {
        default: ANNOTATION_POINTS,
      },
      stubs: DEFAULT_STUBS,
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with customised annotation points', async () => {
    const wrapper = mount(Annotation, {
      propsData: {
        src: './annotateSampleImage.png',
      },
      slots: {
        default: CUSTOMISED_ANNOTATION_POINTS,
      },
      stubs: DEFAULT_STUBS,
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with markdown in header, content and label', async () => {
    const wrapper = mount(Annotation, {
      propsData: {
        src: './annotateSampleImage.png',
      },
      slots: {
        default: MARKDOWN_ANNOTATION_POINTS,
      },
      stubs: DEFAULT_STUBS,
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
});
