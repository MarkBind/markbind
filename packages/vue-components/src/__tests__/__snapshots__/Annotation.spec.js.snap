// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation with customised annotation points 1`] = `
<div
  class="annotate-image-wrapper"
>
  <img
    class="annotate-image"
    height=""
    loading="eager"
    src="./annotateSampleImage.png"
    width=""
  />
   
  <div
    style="top: 0px; left: 0px; height: 0px;"
  >
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <div
                  style="color: red;"
                >
                  Hi
                </div>
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <span
                  class="badge bg-primary"
                >
                  Primary
                </span>
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Annotation with different visual annotation points 1`] = `
<div
  class="annotate-image-wrapper"
>
  <img
    class="annotate-image"
    height=""
    loading="eager"
    src="./annotateSampleImage.png"
    width=""
  />
   
  <div
    style="top: 0px; left: 0px; height: 0px;"
  >
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 60px; height: 60px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: red; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.7; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: black; opacity: 1; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: white;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 30px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Annotation with markdown in header, content and label 1`] = `
<div
  class="annotate-image-wrapper"
>
  <img
    class="annotate-image"
    height=""
    loading="eager"
    src="./annotateSampleImage.png"
    width=""
  />
   
  <div
    style="top: 0px; left: 0px; height: 0px;"
  >
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <!---->
       
      <div
        class="annotate-point"
      >
        <div
          class="popover-annotation"
        >
          <div
            style="position: absolute; pointer-events: all;"
          >
            <div
              class="v-popper v-popper--theme-dropdown"
            >
              <div
                class="hover-wrapper"
              >
                <button
                  class="hover-point"
                  style="background-color: green; opacity: 0.3; width: 40px; height: 40px; cursor: default;"
                />
                 
                <div
                  class="hover-label"
                  style="font-size: 14px; color: black;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
